File: leetcode/4. 寻找两个正序数组的中位数/Solution.go
package _test

func findMedianSortedArrays(nums1 []int, nums2 []int) float64 {
	len1, len2 := len(nums1), len(nums2)
	left, right := -1, -1
	sumLen := len1 + len2

	start1, start2 := 0, 0

	for i := 0; i <= sumLen/2; i++ {
		left = right
		if start1 < len1 && (start2 >= len2 || nums1[start1] < nums2[start2]) {
			right = nums1[start1]
			start1++
		} else {
			right = nums2[start2]
			start2++
		}
	}

	if sumLen&1 == 0 {
		return (float64(left) + float64(right)) / 2
	}

	return float64(right)

}

File: leetcode/4. 寻找两个正序数组的中位数/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    double findMedianSortedArrays(vector<int> &nums1, vector<int> &nums2)
    {
        int len1 = nums1.size();
        int len2 = nums2.size();

        int sumlen = len1 + len2;

        int left = -1, right = -1;
        int start1 = 0, strat2 = 0;

        for (int i = 0; i < sumlen / 2; i++)
        {
            left = right;
            if (start1 < len1 && (strat2 >= len2 || nums1[start1] < nums2[strat2]))
            {
                right = nums1[start1++];
            }
            else
            {
                right = nums2[strat2++];
            }
        }

        if ((sumlen & 1) == 0)
        {
            return (left + right) / 2.0;
        }
        else
        {
            return right;
        }
    }
};